llvm/lib/Transforms/Utils/ScalarEvolutionExpander.cpp

```diff
@@ -175,6 +176,16 @@ SCEVExpander::findInsertPointAfter(Instruction *I,
   return IP;
 }
 
+void SCEVExpander::eraseDeadInstructions(Value *Root) {
+  for (Instruction *I : getAllInsertedInstructions()) {
```

Comment:
Probably makes sense to iterate in `reverse()`, as a later dead instruction may render an earlier one dead as well.